misk/src/main/kotlin/misk/web/interceptors/MetricsInterceptor.kt

```diff
@@ -0,0 +1,30 @@
+package misk.web.interceptors
+
+import misk.Action
+import misk.Chain
+import misk.Interceptor
+import misk.metrics.Metrics
+import misk.metrics.MetricsScope
+import misk.web.Response
+import javax.inject.Inject
+import javax.inject.Singleton
+
+internal class MetricsInterceptor internal constructor(val scope: MetricsScope) : Interceptor {
+
+  @Singleton
+  class Factory @Inject constructor(val m: Metrics) : Interceptor.Factory {
+    override fun create(action: Action): Interceptor? =
+        MetricsInterceptor(m.scope("web.${action.name}"))
+  }
+
+  override fun intercept(chain: Chain): Any? {
+    scope.counter("requests").inc()
+    val result = scope.timer("timing").time { chain.proceed(chain.args) }
+    if (result is Response<*>) {
```

Comment:
Why do we only want to record the status code if the result is a `Response<*>`?

I think we might want this interceptor to run after the `BoxResponseInterceptor`, so we don't ignore requests where we just return a value. See the `HelloWebPostAction` in the exemplar Kotlin app for a case we don't return a `Response` from an action.